frontend/src/component/pages/HomePage.tsx

```diff
@@ -345,65 +308,12 @@ export class HomePage extends Component<Props, State> {
     this.updateCurrentPagedShortLinks(offset, pageSize);
   };
 
-  render = () => {
-    return (
-      <div className="home">
-        {this.state.shouldShowPromo && <ExtPromo />}
-        <Header
-          uiFactory={this.props.uiFactory}
-          onSearchBarInputChange={this.handleSearchBarInputChange}
-          autoCompleteSuggestions={this.state.autoCompleteSuggestions}
-          shouldShowSignOutButton={this.state.isUserSignedIn}
-          shouldShowAdminButton={this.state.shouldShowAdminButton}
-          onSignOutButtonClick={this.handleSignOutButtonClick}
-          onAdminButtonClick={this.handleAdminButtonClick}
-        />
-        <div className={'main'}>
-          <CreateShortLinkSection
-            uiFactory={this.props.uiFactory}
-            ref={this.createShortLinkSection}
-            longLinkText={this.state.longLink}
-            alias={this.state.alias}
-            shortLink={this.state.shortLink}
-            inputErr={this.state.inputErr}
-            createdUrl={this.state.createdUrl}
-            qrCodeUrl={this.state.qrCodeUrl}
-            isShortLinkPublic={this.state.isShortLinkPublic}
-            onLongLinkTextFieldBlur={this.handleLongLinkTextFieldBlur}
-            onLongLinkTextFieldChange={this.handleLongLinkChange}
-            onShortLinkTextFieldBlur={this.handleCustomAliasTextFieldBlur}
-            onShortLinkTextFieldChange={this.handleAliasChange}
-            onPublicToggleClick={this.handlePublicToggleClick}
-            onCreateShortLinkButtonClick={this.handleCreateShortLinkClick}
-          />
-          {this.state.isUserSignedIn && (
-            <div className={'user-short-links-section'}>
-              {this.props.uiFactory.createUserShortLinksSection({
-                onPageLoad: this.handleOnShortLinkPageLoad,
-                pagedShortLinks: this.state.currentPagedShortLinks
-              })}
-            </div>
-          )}
-        </div>
-        <Footer
-          uiFactory={this.props.uiFactory}
-          onShowChangeLogBtnClick={this.handleShowChangeLogBtnClick}
-          authorName={'Harry'}
-          authorPortfolio={'https://github.com/byliuyang'}
-          version={this.props.versionService.getAppVersion()}
-        />
-        <ChangeLogModal
-          ref={this.changeLogModalRef}
-          onModalOpen={this.handleOpenChangeLogModal}
-          changeLog={this.state.changeLog}
-          defaultVisibleLogs={3}
-        />
-
-        <SignInModal ref={this.signInModal} uiFactory={this.props.uiFactory} />
-        <ErrorModal store={this.props.store} />
+  private handleOnShortLinkCreated = (shortLink: string) => {
+    this.copyShortenedLink(shortLink);
+    this.refreshUserShortLinks();
+  };
 
-        <Toast ref={this.toastRef} />
-      </div>
-    );
+  private handleOnRequestSignIn = () => {
```

Comment:
Do you think we should rename this to `authenticationErrHandler`? We will let the home page decide what happens when the user is not authenticated.

Reply:
Done.